src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/UnusedTracker.groovy

```diff
@@ -12,18 +12,32 @@ import org.vafer.jdependency.Clazz
 import org.vafer.jdependency.Clazzpath
 import org.vafer.jdependency.ClazzpathUnit
 
+import java.nio.file.Path
+
 /** Tracks unused classes in the project classpath. */
 class UnusedTracker {
-    private final FileCollection toMinimize
-    private final List<ClazzpathUnit> projectUnits
-    private final Clazzpath cp = new Clazzpath()
+    protected final FileCollection toMinimize
+    protected final List<ClazzpathUnit> projectUnits
+    protected final Clazzpath cp = new Clazzpath()
 
-    private UnusedTracker(List<File> classDirs, FileCollection classJars, FileCollection toMinimize) {
+    protected UnusedTracker(List<File> classDirs, FileCollection classJars, FileCollection toMinimize) {
         this.toMinimize = toMinimize
         projectUnits = classDirs.collect { cp.addClazzpathUnit(it) }
         projectUnits.addAll(classJars.collect { cp.addClazzpathUnit(it) })
     }
 
+    boolean performsFullShrinking() {
```

Comment:
None of these methods are necessary in this class since they aren't used by it. If they are only needed by the R8 implementation, then they should reside in that class.